dev/connection_holder.h

```diff
@@ -45,7 +44,7 @@ namespace sqlite_orm {
 
           protected:
             sqlite3* db = nullptr;
-            int _retain_count = 0;
+            std::atomic_int _retain_count = 0;
```

Comment:
Well... it's not like there's a huge toll in terms of performance. It's just a bit of memory fencing somewhere at the CPU level.
However, if it's easy to make it conditional, then why not.

Reply:
If we really want to be pedantic we could optimize on the memory order. But I would need to dive into this again.

Reply:
@trueqbit what runtime multithreading configuration r u talking about?

Reply:
@fnc12 You can call [sqlite3_config()](https://www.sqlite.org/c3ref/config.html) with [SQLITE_CONFIG_SINGLETHREAD /SQLITE_CONFIG_MULTITHREAD / SQLITE_CONFIG_SERIALIZED](https://www.sqlite.org/c3ref/c_config_covering_index_scan.html) at start or runtime.

Reply:
we need to solve what to do to help you and other devs who faces the same issue. The options I see:
1) just do what Klaus did in the very first commit of this PR: replace `int` with `std::atomic<int>`. 
    *Pros*: easy and can be committed today. 
    *Cons*: some devs who uses `sqlite_orm` in single-threaded mode and require high performance will not like this solution cause it will add a multi-thread management entity which is not desirable. I know such devs. The amount of them is low but they exist.
2) add `#ifdef SQLITE_THREADSAFE std::atomic` with `#else int`.
    *Pros*: will solve the issue with single-threaded devs
    *Cons*: it's not a perfect solution cause thread safety can be specified with runtime functions also.
3) put your option here...

IMHO second option is good. A lot of devs posted issues about thread safety and most of them defined `SQLITE_THREADSAFE` for it. So I think we can easily make `#ifdef SQLITE_THREADSAFE std::atomic`. If someone appears who doesn't have `SQLITE_THREADSAFE` defined but enables multi-threading using runtime (never met such a developer) we'll tell him/her what to do. I am sure there will not be any problem of defining `SQLITE_THREADSAFE`.

I need your opinions guys.

Also @trueqbit I'd add a unit test for this case to eliminate an opportunity to break this bugfix.

Reply:
I agree that we need a well-defined solution. But we are arguing about high-performance in the absolutely wrong place. How come you would assume it takes a huge toll on performance? Where do you see the huge difference between `++int` and `++atomic_int`? At the end of the day it's just a memory order instruction at the CPU level.

Your 2nd option doesn't work unless we define a macro for sqlite_orm itself - `SQLITE_THREADSAFE` is a flag specified at compile time and isn't available later on. Making the build dependent on yet another option is much harder to get right than sacrificing 2 femtoseconds of performance.

As for the unit test I don't see a good possibility to create one, because the whole procedure of retaining and releasing is encapsulated. Even creating an artificial test case is hard to do. It would be more for showcasing, which can be done at the documentation level.

Reply:
Ok sounds reasonable. Let's add `std::atomic` with no condition and no unit test.

BTW: what is `femtoseconds`? Feminist to second? 😀

Reply:
femto is a quadrillionth of a second. I probably should have taken yocto, instead :)